src/Survey/ItechArt.Survey.WebApp/ViewModels/RegistrationViewModel.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace ItechArt.Survey.WebApp.ViewModels;
+
+public class RegistrationViewModel
+{
+    [StringLength(30, MinimumLength = 3, ErrorMessage = "User name must consist of 3-30 symbols")]
+    [RegularExpression(
+       @"^(?=.{3,30}$)(?![_0-9\s])[a-zA-ZА-Яа-я0-9\s_]+(?<![_\s])$",
```

Comment:
let's move min/max, regexes, etc. to static class with constans and reuse it here and in options configuration